pre-piksi-multi-launch-updates.md

```diff
@@ -0,0 +1,222 @@
+
+These changes attempt to accomplish a few things:
+- Make messages more bandwidth efficient
+- Make user side logic of using messages more obvious
+- Remove state from the protocol, where possible
+
+### 1. Send solution messages at all times, regardless of fix status
+
+- Affects the following messages:
+   - GPS time
+   - DOPS (to add)
+   - Pos ECEF
+   - Pos LLH
+   - Baseline ECEF
+   - Baseline NED
+   - Vel ECEF
+   - Vel NED
+   - Baseline heading 
+- Update flags 
+   - Describe "fix mode" portion of flags field to be for each individual message only
+   - in all messages, 0 corresponds to "invalid"
+   - RTK Float and RTK Fixed flags have same numerical value across different messages where relevant
+   - No separate flags for simulation mode
+   - Only a minor release without a change to MSG_ID can be used to assign a meaning to reserved flags
+- Flags detail
+    - POS flags:
+       - Bits 2-0
+         - 000 Invalid
+         - 001 RTK Float
+         - 010 RTK Fixed
+         - 011 SPP
+         - 100 Reserved
+         - 101 Reserved
+         - 110 Reserved
+         - 111 Reserved
+       - Bits 6-3
+         - Reserved
+       - Bit 7
+         - 0 No RAIM repair
+         - 1 RAIM repair
+       - Note, we remove MSL vs Ellipsoid height as we have nothing but ellipsoid currently
+
+    - BASELINE flags:
+      - Bits: 76543210
+      - Bits 2-0:
+        - 000 Invalid
+        - 001 RTK Float
+        - 010 RTK Fixed
+        - 011 Reserved
+        - 100 Reserved
+        - 101 Reserved
+        - 110 Reserved
+        - 111 Reserved
+      - Bits 6-3:
+        - Reserved
+      - Bit 7:
+        - 0 No RAIM Repair
+        - 1 RAIM repair
+
+     - VEL flags:
+        - Bits: 76543210
+        - Bits 2-0:
+          - 000 Invalid
+          - 001 Measured Doppler Derived
+          - 010 Computed Doppler Derived
+          - 011 Reserved
+          - 100 Reserved
+          - 101 Reserved
+          - 110 Reserved
+          - 111 Reserved
+        - Bits 6-3:
+          - Reserved
+        - Bit 7:
+          - 0 No RAIM Repair
+          - 1 RAIM repair
+
+    - GPS_TIME flags:
+      - Bits: 76543210
+      - Bits 2-0:
+        - 000 Invalid
+        - 001 PVT derived
+        - 010 Reserved (will be filled in when RTC comes online and maybe with coarse time)
+        - 011 Reserved
+        - 100 Reserved
+        - 101 Reserved
+        - 110 Reserved
+        - 111 Reserved
+      - Bits 6-3:
+        - Reserved
+      - Bits 7:
+        - 0 No RAIM repair
+        - 1 RAIM repair
+
+Heading:
+Deprecate for the time being
+
+Dops:
+DOPS calculation needs to update when we have RTK solution or RAIM flag to represent the satellites actually used.  When in RTK mode of any kind, DOPS will represent the DOPS from the RTK solution.  If no RTK solution is available (pseudo absolute or baseline), DOPS will be only SPP DOPS.  Flags will follow the position message definition.
```

Comment:
I've brought this up for obs before - but this kind of modal message (dependent on the solution type) is funky to use for end users.